timelock-agent/src/main/java/com/palantir/timelock/paxos/TimelockPaxosAcceptorAdapter.java

```diff
@@ -63,9 +65,23 @@ public long getLatestSequencePreparedOrAccepted() {
      */
     public static Function<Client, List<PaxosAcceptor>> wrap(
             PaxosUseCase paxosUseCase,
-            List<TimelockPaxosAcceptorRpcClient> acceptors) {
-        return client -> acceptors.stream()
-                .map(acceptor -> new TimelockPaxosAcceptorAdapter(paxosUseCase, client, acceptor))
-                .collect(Collectors.toList());
+            PaxosRemoteClients remoteClients) {
+        switch (paxosUseCase) {
+            case LEADER_FOR_ALL_CLIENTS:
+                return _client -> remoteClients.singleLeaderAcceptor().stream()
+                        .<PaxosAcceptor>map(Function.identity())
+                        .collect(Collectors.toList());
+            case LEADER_FOR_EACH_CLIENT:
```

Comment:
These paxos use cases sounds surreal, but I trust it makes sense

Reply:
I would have put this logic inside the `PaxosUseCase`, but dependencies :'(